### PR TITLE
Add MobileApp and CPM campaign controls

### LINKED ISSUE #296: campaigns add/update: MobileAppCampaign and CpmBannerCampaign parity
Parent: #250

## Scope
Add canonical typed flags for MobileAppCampaign and CpmBannerCampaign optional parity gaps.

## CLI Contract
- Use canonical typed CLI flags only.
- Do not add public `--json` input or blob payload forwarding.
- Keep README/help examples single-line.
- Validate only documented local constraints unless this child issue explicitly says otherwise.

## WSDL Audit Routing
campaigns.add/update rows under MobileAppCampaign and CpmBannerCampaign.

## Acceptance Criteria
- Focused dry-run tests prove the exact Yandex Direct payload shape.
- `tests/test_wsdl_parity_gate.py` routes the matching optional audit rows to this issue or marks them supported.
- `tests/WSDL_OPTIONAL_FIELD_AUDIT.md` is updated consistently.
- Relevant CLI/help/docs tests pass.

## Out Of Scope
- Live API write tests.
- Unrelated subtype blocks from parent #250.
- Broad all-optional-fields parity outside this child issue.

### PR DESCRIPTION
**Pull request by axisrow**
Closes #296.

## Summary
- Adds typed campaign flags for MobileAppCampaign Settings and NegativeKeywordSharedSetIds.
- Adds typed campaign flags for CpmBannerCampaign Settings, CounterIds, FrequencyCap, and VideoTarget.
- Updates WSDL optional-field parity rows and README EN/RU single-line examples.

## Documentation checked
- Yandex Direct campaigns.add MobileAppCampaign docs: Settings and NegativeKeywordSharedSetIds are subtype fields.
- Yandex Direct campaigns.add/update CpmBannerCampaign docs: Settings, CounterIds, FrequencyCap, and VideoTarget are subtype fields.
- BiddingStrategy residual nested coverage stays routed to #290.

## Verification
- python3 -m pytest tests/test_dry_run.py -k 'campaigns and (mobile_app_campaign or cpm_banner_campaign or frequency_cap or tracking_params)'
- python3 scripts/build_wsdl_optional_field_audit.py --check
- python3 -m pytest tests/test_wsdl_parity_gate.py
- python3 -m pytest tests/test_cli_contract.py -k 'json or readme_direct_examples'
- python3 -m pytest tests/test_cli.py tests/test_dry_run.py tests/test_wsdl_parity_gate.py
- mypy .
- git diff --check

### CODE REVIEW

**Comment by axisrow** [comment]
@codex @claude review. Focus on blocking or actionable issues only: correctness against issue #296 scope, Yandex Direct API payload shape, CLI contract, tests, docs, WSDL audit integrity, or CI risk. Do not nitpick style, naming, formatting, or subjective preferences.

**Comment by axisrow** [comment]
Addressed the review feedback in 8b8aaf6: CPM/Mobile strategy customization is rejected in #296 and remains under #290, add help now references CounterIds.Items, FrequencyCap.PeriodDays null is exposed through typed --frequency-cap-period-all, and the help/name tests were clarified. @codex @claude review. Focus on blocking or actionable issues only: correctness against issue #296 scope, Yandex Direct API payload shape, CLI contract, tests, docs, WSDL audit integrity, or CI risk.

File: tests/test_dry_run.py [hunk 1]
```diff
@@ -6336,6 +6336,9 @@ def test_campaigns_help_exposes_text_and_unified_campaign_optional_flags():
         assert "--package-platform-maps" in result.output
         assert "--package-platform-search-organization-list" in result.output
         assert "--negative-keyword-shared-set-ids" in result.output
```
[the remark this answers is not in the document]
Comment (axisrow) [reply tests/test_dry_run.py:6338]: Addressed in 8b8aaf6. The CPM banner help assertions are now split into tests/test_dry_run.py::test_campaigns_help_exposes_cpm_banner_optional_flags, separate from the text/unified optional flag help test.

File: direct_cli/commands/campaigns.py [hunk 2]
```diff
@@ -860,7 +876,8 @@ def get(
     "--counter-ids",
     help=(
         "Comma-separated Metrika counter IDs "
-        "(TextCampaign/UnifiedCampaign/DynamicTextCampaign.CounterIds)"
+        "(TextCampaign/UnifiedCampaign/DynamicTextCampaign/"
+        "CpmBannerCampaign.CounterIds)"
```
[the remark this answers is not in the document]
Comment (axisrow) [reply direct_cli/commands/campaigns.py]: Addressed in 8b8aaf6. The add command help now references CpmBannerCampaign.CounterIds.Items to match the emitted CounterIds.Items payload shape.

File: direct_cli/commands/campaigns.py [hunk 3]
```diff
@@ -1603,6 +1681,42 @@ def add(
...
+                    negative_keyword_shared_set_ids_obj
+                )
+            campaign_data["MobileAppCampaign"] = mobile_campaign
+        elif campaign_type_norm == "CPM_BANNER_CAMPAIGN":
+            cpm_campaign: Dict[str, object] = {
+                "BiddingStrategy": {
+                    "Search": {"BiddingStrategyType": search_strategy or "SERVING_OFF"},
+                    "Network": {
+                        "BiddingStrategyType": network_strategy or "MANUAL_CPM"
+                    },
```
[the remark this answers is not in the document]
Comment (axisrow) [reply direct_cli/commands/campaigns.py:1727]: Addressed in 8b8aaf6. For CPM_BANNER_CAMPAIGN, the add allowed-list no longer includes --search-strategy or --network-strategy, so those flags are rejected before payload construction. Added dry-run coverage in tests/test_dry_run.py::test_campaigns_add_cpm_banner_rejects_bidding_strategy_flags. The remaining internal defaults are fixed values only and the broader typed BiddingStrategy work stays routed to #290.

File: tests/test_dry_run.py [hunk 4]
```diff
@@ -6582,18 +6728,19 @@ def test_campaigns_update_tracking_params_without_type_rejected():
...
-def test_campaigns_update_unsupported_type_rejected():
+def test_campaigns_update_tracking_params_on_unsupported_type_rejected():
     result = _rejected(
         "campaigns",
         "update",
         "--id",
         "123",
         "--type",
-        "MOBILE_APP_CAMPAIGN",
+        "CPM_BANNER_CAMPAIGN",
```
[the remark this answers is not in the document]
Comment (axisrow) [reply tests/test_dry_run.py]: Addressed in 8b8aaf6. The test is now named test_campaigns_update_tracking_params_on_cpm_banner_rejected, matching the actual constraint: CPM_BANNER_CAMPAIGN is supported, but --tracking-params is not compatible with that subtype.

File: direct_cli/commands/campaigns.py [hunk 5]
```diff
@@ -512,6 +513,21 @@ def _build_dynamic_placement_types(
...
+    period_days: Optional[int],
+) -> Optional[dict]:
+    """Build CpmBannerCampaign.FrequencyCap from paired typed flags."""
+    if impressions is None and period_days is None:
+        return None
+    if impressions is None or period_days is None:
+        raise click.UsageError(
+            "--frequency-cap-impressions and --frequency-cap-period-days "
+            "must be provided together"
+        )
```
[the remark this answers is not in the document]
Comment (axisrow) [reply direct_cli/commands/campaigns.py]: Addressed in 8b8aaf6. FrequencyCap.PeriodDays null is now expressible through the typed --frequency-cap-period-all flag, with add/update dry-run tests covering the emitted PeriodDays: null payload. Reset/null blob semantics remain out of scope except for this explicit typed flag.
